membrane_curvature/base.py

```diff
@@ -0,0 +1,72 @@
+import numpy as np
+
+from membrane_curvature.lib.mods import derive_surface, mean_curvature, gaussian_curvature
+import numpy as np
+
+
+def __init__(self, universe, select,
+             x_bins=100, y_bins=100,
+             x_range=None,
+             y_range=None,
+             pbc=True):
+    self.universe = universe
+    self.selection = select
+    self.pbc = pbc
+    self.y_bins = y_bins
+    self.x_bins = x_bins
+    self.x_range = x_range if x_range else (0, universe.dimensions[0])
+    self.y_range = y_range if y_range else (0, universe.dimensions[1])
+
+    # Warns if range doesn't cover entire dimensions of simulation box
+    if (x_range < universe.dimensions[0]) or (y_range < universe.dimensions[1]):
+        raise ValueError("Minimum range must be ({}, {})").format(self.universe.dimensions[0],
+                                                                  self.universe.dimenions[1])
+
+
+def _prepare(self):
+
+    # Initialize empty np.array(cumulative, count) of results
+    cumulative = np.zeros((self.x_bins, self.y_bins))
+    count = np.zeros((self.x_bins, self.y_bins))
+
+    self.results.surface = (cumulative, count)
+    self.results.mean = (cumulative, count)
+    self.results.gaussian = (cumulative, count)
```

Comment:
```suggestion
    self.results.surface = np.zeros((self.x_bins, self.y_bins))
    self.results.surface_count = np.zeros((self.x_bins, self.y_bins))
    self.results.mean = np.zeros((self.x_bins, self.y_bins))
    self.results.mean_count = np.zeros((self.x_bins, self.y_bins))
    self.results.gaussian = np.zeros((self.x_bins, self.y_bins))
    self.results.gaussian_count = np.zeros((self.x_bins, self.y_bins))
```
I think these results would be easier to work with if they were separate.

You also need to construct new arrays each time. As it is, currently, `self.results.gaussian[0] is self.results.mean[0]` will return True -- they are the same object in memory. That means, when you call `_output` on the surface, mean, and gaussian results, you are updating the same arrays 3 times.

Reply:
Actually, I was about to edit when I received the review 😄. 

I had it initially in the explicit form as you suggested, but I found it ugly (long and redundant). Then I changed to the submitted version, but yes, that's incorrect. I will submit a change with the explicit version. 

What do you think about this option?

```
cumulative = np.zeros((self.x_bins, self.y_bins))
count = np.zeros((self.x_bins, self.y_bins))

self.results.surface = (cumulative.copy() , count.copy() )
self.results.mean = (cumulative.copy(), count.copy() )
self.results.gaussian = (cumulative.copy(), count.copy())
```

Reply:
I still wouldn't be keen on having a tuple of arrays in the results. Simpler is often better, even if it results in more lines of code.

Reply:
Thanks for the comment @lilyminium. Maybe it's something silly, but may I ask why tuples of arrays are not the best approach here?

Reply:
I'm mostly thinking of the user experience when I say this. If I wanted to work with the raw `mean` array, it's much handier for me to work with `self.results.mean_array` than `self.results.mean[0]`. The `[0]` might look like a small addition but it means that I have to remember which order the arrays are in (because it might be `self.results.mean[1]`), so I have to look up the documentation each time (in fact this is a real downside of the existing RMSD class in MDAnalysis -- the actual RMSD results are only the last three columns of the `self.results.rmsd` array).

It's also easier to document `self.results.mean_array is the array of the mean surfaces` instead of `self.results.mean is a tuple of arrays, the first array is the array of mean surfaces, the second is the array of counts` :)

Anyway, I think you may be able to get rid of the counter arrays by using `np.nanmean`.

Reply:
I see. Thanks for the clear explanation. 
The question was more out of curiosity to see the downside of the approach I used here. And to be honest, while writing code I never think of that, but as an user I can totally see it :) Very helpful, thanks again.

Reply:
It was a good question! There's always a balance between making life easier for the developer but thinking about users too :p